### PR TITLE
#5408: add Datadog for performance monitoring/tracing

### LINKED ISSUE #5408: Add RUM for performance monitoring
Context
---
- We've switched to Datadog for APM/ASM on the backend
- They have a RUM product for front-end monitoring

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #5408 
- Is behind the `telemetry-performance` feature flag
- Monitors the following:
   - Extension Console
   - Page Editor
   - Sidebar  

## Discussion

- I left the default `sessionSampleRate` and `sessionReplaySampleRate`. We'll need to determine what the rates for these need to be

## Release QA

- [ ] Ensure the tracing headers are being processed by the server (CORS might be blocking them)

## Demo

Data in Datadog
image

Tracing headers included with requests:
image

## Future Work

- Modify the Page Editor to either: 1) use hash routing, 2) mark page switches to better support navigation timings
- Add action tracking by hooking into report event: https://docs.datadoghq.com/real_user_monitoring/browser/tracking_user_actions/
- Consider adding RUM for contentPage startup and tiny pages (modal, etc.)
- Upload source maps to Datadog to improve stack trace analysis

## Checklist

- [x] Add tests: N/A
- [x] Designate a primary reviewer: @johnnymetz

### CODE REVIEW

File: src/options/pages/activateExtension/ActivatePage.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ const ActivatePage: React.FunctionComponent = () => {
     data: extension,
     isLoading,
     error,
-  } = useFetch<CloudExtension>(`/api/extensions/${extensionId}`);
+  } = useFetch<CloudExtension>(`/api/extensions/${extensionId}/`);
```
Comment (twschiller) [inline src/options/pages/activateExtension/ActivatePage.tsx:39]: Unrelated change - I noticed this was causing an unnecessary redirect (to add the missing slash)

**Review by johnnymetz — APPROVED** [review]
